### PR TITLE
add documentation for setting custom headers

### PR DESCRIPTION
**Pull request by foosinn**
Currently there is no documentation for setting custom headers, but its mentioned in the sample configuration file, see [sample config](https://github.com/keycloak/keycloak-gatekeeper/blob/fa53727277a1f931498fddb998bb54149845bf23/config_sample.yml#L45)

### CODE REVIEW

**Review by matthewhelmke — APPROVED** [review]
Thank you very much! Gatekeeper is something new and that I have been working to clean up. I had not gotten around to writing this yet and appreciate that you saw the need and submitted this.

A word on our process: we have two reviews for all docs PRs. One is from me from a writing standpoint. I have reviewed this and am happy with what you have submitted (thanks again!). The second will be from someone in engineering who will check the technical details again before we merge.

**Comment by abstractj** [comment]
Thanks @foosinn, created a Jira to track that https://issues.jboss.org/browse/KEYCLOAK-8986

**Comment by abstractj** [comment]
@foosinn @matthewhelmke I would like to put this PR on hold and let me try to explain why. The documentation written by @foosinn seems accurate for me, although, we may have a bug in Gatekeeper, because setting custom headers at least for me here, seems to have no practical effect.

If we merge this change without fixing it on Gatekeeper first, the information can be misleading.

@foosinn out of curiosity have you managed to inject the custom headers using `--headers`? Or you submitted this changed based on YAML files?

**Comment by foosinn** [comment]
Hey @abstractj i did use the yaml configuration for production, but i tested both and it worked. 

**Note:**

The headers in the sample config did not work for me. i asumed the `_` character is not supported in html headers names, but i',m not much of a web developer.

```
headers:
  myheader_name: my_header_value
```

**Comment by abstractj** [comment]
@foosinn so running something like the code below is working for you?

```
bin/keycloak-gatekeeper --client-id=py-rest --client-secret="$SECRET" \
--discovery-url=http://$KCIP:8180/auth/realms/master \
--enable-default-deny=true \
--encryption-key=$ENCRYPTION_KEY \
--listen=$APPIP:4000 \
--redirection-url=http://$APPIP:4000 \
--upstream-url=http://$APPIP:5000 \
--resources='uri=/*' \
--secure-cookie=false \
--headers="myheader=header1"
```

Locally to me things don't break, but at the same time, seems like there's no practical effect. The reason why I'm asking, is just make sure that this feature is working for everyone trying the docs.

**Comment by foosinn** [comment]
Works for me:

```sh
./gatekeeper \
  --client-id=foo_test \
  --client-secret="f6233ca7-845d-481c-8916-5ab57e7f1aeb" \
  --discovery-url=http://localhost:8080/auth/realms/master \
  --encryption-key=B7pTUNy7kWL78TCy1t6sxUrMUs9SmBMR \
  --listen=:8001 \
  --redirection-url=http://localhost:8001 \
  --upstream-url=http://localhost:5000 \
  --secure-cookie=false \
  --headers="test=test"
```

as server i used a simple python flask app:
`app.py`:
```python
from flask import Flask, request, Response
app = Flask(__name__)

@app.route("/")
def hello():
    response = '\n'.join('%s: %s' % i for i in request.headers.items())
    response = Response(response, mimetype='text/plain')
    return response
```
`FLASK_APP=app flask run`

Result:

```
Host: localhost:5000
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.80 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Language: en-US,en;q=0.9,de;q=0.8
Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR...
Cache-Control: max-age=0
Cookie: kc-access=eyJhbGciOiJSUzI1NiIsIn...
Dnt: 1
Test: test
Upgrade-Insecure-Requests: 1
X-Auth-Audience: foo_test
X-Auth-Email: 
X-Auth-Expiresin: 2018-12-14 11:26:20 +0000 UTC
X-Auth-Groups: 
X-Auth-Roles: create-realm,...
X-Auth-Subject: bd818eca-65e5-4b5c-ad79-6117b5ff6702
X-Auth-Token: eyJhbGciOiJSUzI1NiIsInR5cCIgO...
X-Auth-Userid: foo
X-Auth-Username: foo
X-Forwarded-For: ::1
X-Forwarded-Host: localhost:8001
X-Forwarded-Proto: 
Accept-Encoding: gzip
```

The header is set: `Test: test`

**Comment by abstractj** [comment]
@foosinn you're damn right. Thanks for all the details.

@matthewhelmke marked it as ready. So we can merge it, in the next release.

**Review by abstractj — APPROVED** [review]

**Pull request merged**
